sycl/include/sycl/accessor.hpp

```diff
@@ -2215,8 +2219,78 @@ class __SYCL_SPECIAL_CLASS accessor<DataT, Dimensions, AccessMode,
     return local_ptr<DataT>(getQualifiedPtr());
   }
 
-  bool operator==(const accessor &Rhs) const { return impl == Rhs.impl; }
-  bool operator!=(const accessor &Rhs) const { return !(*this == Rhs); }
+  bool operator==(const local_accessor_base &Rhs) const {
+    return impl == Rhs.impl;
+  }
+  bool operator!=(const local_accessor_base &Rhs) const {
+    return !(*this == Rhs);
+  }
+};
+
+template <typename DataT, int Dimensions, access::mode AccessMode,
```

Comment:
Could you please add a TODO here saying that it is deprecated through `access::target::local` and should be removed?

Reply:
Done